app/models/piece.rb

```diff
@@ -158,6 +139,22 @@ def is_white?(piece)
  		piece.color == "white" ? true : false
  	end	
 
+ 	def puts_in_check?(x,y)
+		if color == "white"
+			opposite_color = "black"
+		elsif color == "black"
+			opposite_color = "white"
+		end
+				
+		opponents = game.pieces.where(color: opposite_color)
+		opponents.each do |piece|
+			if piece.valid_move?(x,y) 
+				return true 
+			else
+		  	return false
+		  end
+		end
+	end
 end
 
```

Comment:
I think there is a flaw here. From what I see, if the first opponent doesn't cause a check, it will return false. so essentially the loop ends after the first piece. What you want is a break statement and the return false right at the bottom of the method. does that make sense?